code/services/LDAPService.php

```diff
@@ -592,6 +616,205 @@ public function updateGroupFromLDAP(Group $group, $data)
     }
 
     /**
+     * Creates a new LDAP user from the passed Member record.
+     *
+     * @param Member $member
+     */
+    public function createLDAPUser(Member $member)
+    {
+        if (!$this->enabled()) {
+            return;
+        }
+
+        // Normalise username to lowercase to ensure we don't have duplicates of different cases
+        $member->Username = strtolower($member->Username);
+
+        // Create user in LDAP using available information.
+        $dn = sprintf('CN=%s,%s', $member->Username, LDAP_NEW_USERS_DN);
+
+        try {
+            $this->add($dn, array(
+                'objectclass' => 'user',
+                'cn' => $member->Username,
+                'accountexpires' => '9223372036854775807',
+                'useraccountcontrol' => '66048',
+                'userprincipalname' => sprintf(
+                    '%s@%s',
+                    $member->Username,
+                    $this->gateway->config()->options['accountDomainName']
+                ),
+            ));
+        } catch (\Exception $e) {
+            throw new ValidationException('LDAP synchronisation failure: '.$e->getMessage());
+        }
+
+        $user = $this->getUserByUsername($member->Username);
+        if (empty($user['objectguid'])) {
+            throw new ValidationException('LDAP synchronisation failure: user missing GUID');
+        }
+
+        // Creation was successful, mark the user as synced with LDAP.
+        $member->GUID = $user['objectguid'];
+        $member->IsImportedFromLDAP = 1;
+    }
+
+    /**
+     * Update the Member data back to the corresponding LDAP user object.
+     *
+     * @param Member $member
+     * @throws ValidationException
+     */
+    public function updateLDAPFromMember(Member $member)
+    {
+        if (!$this->enabled()) {
+            return;
+        }
+        if (!$member->IsImportedFromLDAP) {
+            return;
+        }
+
+        $data = $this->getUserByGUID($member->GUID);
+        if (empty($data['objectguid'])) {
+            throw new ValidationException('LDAP synchronisation failure: user missing GUID');
+        }
+
+        $dn = $data['distinguishedname'];
+
+        // Normalise username to lowercase to ensure we don't have duplicates of different cases
+        $member->Username = strtolower($member->Username);
+
+        try {
```

Comment:
if $this->move or $this->update fails, we can't easily tell because the try .. catch wraps them both.